docs/DeveloperGuide.md

```diff
@@ -1,34 +1,152 @@
 # Developer Guide
+* [Setting up](#1-setting-up)
+* [Design](#2-design)
+* [Implementation](#3-implementation)
+  * [Undo features](#31-proposed-undo-feature)
+    * [View By Category](#311-view-by-category)
+* [Documentation](#4-documentation)
+* [Testing](#5-testing)
+* [Dev Ops](#6-dev-ops)
+  * [Appendix A: Product Scope](#61-appendix-aproduct-scope)
+  * [Appendix B:Value proposition](#62-appendix-bvalue-proposition)
+  * [Appendix C:User Stories](#63-appendix-cuser-stories)
+  * [Appendix D:Non-Functional Requirements](#64-appendix-dnon-functional-requirements)
+  * [Appendix E:Glossary](#65-appendix-eglossary)
+  * [Appendix F:Instructions for Manual Testing](#66-appendix-finstructions-for-manual-testing)
+## 1. Setting up
+## 2. Design
 
-## Design & Implementation
+{Describe the design of the product. Use **Architecture Diagram** which has not been covered yet.}
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+##### 3.1.1.1 Proposed Implementation
+The list by category mechanism is facilitated by ListCommand which extends Command.
 
+Given below is an example usage scenario and how the mechanism behaves at each step.
 
-## Product Scope
-### Target user profile
+Step1: The user adds a task:  `add n/2113 c/DEADLINE d/2020-03-30 t/11:59` to add a task.
+Since the user adds a new category `DEADLINE`, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
 
-{Describe the target user profile}
+Step2: The users adds a task: `add n/3145 c/CLASS` to add a class. 
+Since the user adds a new category `CLASS`, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
 
-### Value proposition
+Step3: Users list the class he just added by category CLASS: `list c/CLASS`. The TaskList will
+return the index of the tasks based on the categoryMap. Then the task in the `CLASS` category
+will be displayed. 
 
-{Describe the value proposition: what problem does it solve?}
+This is the Sequence Diagram for list by category. 
+![Sequence Diagram for View](images/listCategorySequence.png)
```

Comment:
can try to get rid of the watermark here, it is a bit disturbing:
![image](https://user-images.githubusercontent.com/9150683/77991085-880a1d00-7355-11ea-80d9-64529343e88e.png)